packages/amplify-graphql-relational-transformer/src/__tests__/amplify-graphql-relations-with-custom-primary-key.test.ts

```diff
@@ -0,0 +1,474 @@
+import { PrimaryKeyTransformer } from '@aws-amplify/graphql-index-transformer';
+import { ModelTransformer } from '@aws-amplify/graphql-model-transformer';
+import { GraphQLTransform, validateModelSchema } from '@aws-amplify/graphql-transformer-core';
+import { parse } from 'graphql';
+import { BelongsToTransformer, HasManyTransformer, HasOneTransformer } from '..';
+
+describe('custom primary key and relational directives', () => {
+  it('uses primary key name for hasOne relational connection field', () => {
+    const inputSchema = `
+      type Patient @model {
+        ssn: ID! @primaryKey
+        givenName: String
+        familyName: String
+      }
+
+      type MedicalAppointment @model {
+        appointmentReference: ID! @primaryKey
+        patient: Patient @hasOne
+        provider: String
+        date: AWSDateTime
+      }
+    `;
+
+    const transformer = new GraphQLTransform({
+      featureFlags: {
```

Comment:
To increase your Signal to Noise on these tests, I'd recommend creating a helper to create mock feature flags with that accepts a boolean for useFieldNameForPrimaryKeyConnectionField instead of constructing for each test.